src/aegis/src/aegis_server.erl

```diff
@@ -120,7 +128,7 @@ decrypt(#{} = Db, Key, Value) when is_binary(Key), is_binary(Value) ->
                 {error, Reason} ->
                     erlang:error(Reason)
             end;
-        false ->
+        _ ->
             process_flag(sensitive, true),
```

Comment:
shouldn't we evict the entry here?

Reply:
Good call! I've added it in `insert_key` handler called from `do_open_db` to avoid second gen_call.